Formula/libobjc2.rb

```diff
@@ -0,0 +1,57 @@
+class Libobjc2 < Formula
+  desc "Objective-C runtime library intended for use with Clang"
+  homepage "https://github.com/gnustep/libobjc2"
+  url "https://github.com/gnustep/libobjc2/archive/refs/tags/v2.1.tar.gz"
+  sha256 "78fc3711db14bf863040ae98f7bdca08f41623ebeaf7efaea7dd49a38b5f054c"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :test
+  # Clang explicitly forbids building Mach-O binaries of libobjc2.
+  # https://reviews.llvm.org/D46052
+  # macOS provides an equivalent Objective-C runtime.
+  depends_on :linux
+
+  # While libobjc2 is built with clang, it does not use any LLVM runtime libraries.
+  uses_from_macos "llvm" => [:build, :test]
+
+  # Clang must be used on Linux because GCC Objective-C support is insufficient.
+  fails_with :gcc
+
+  resource "robin-map" do
```

Comment:
Any value in adding `robin-map` as a separate formula? No strong opinion on it but wondering if it's been considered.